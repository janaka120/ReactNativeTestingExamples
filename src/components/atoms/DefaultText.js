// @flow
import * as React from 'react';
import {Text, StyleSheet} from 'react-native';

import {scaleFontWithLineHeight} from '_styles/mixins';
import {SECONDARY} from '_styles/colors';

type Props = {
  customStyle?: {},
  text: string,
};
const DefaultText = ({customStyle, text}: Props) => {
  return (
    <Text
      multiline={true}
      numberOfLines={5}
      style={[styles.textStyle, customStyle]}>
      {text}
    </Text>
  );
};
export default DefaultText;

const styles = StyleSheet.create({
  textStyle: {
    ...scaleFontWithLineHeight(16),
    color: SECONDARY,
  },
});
