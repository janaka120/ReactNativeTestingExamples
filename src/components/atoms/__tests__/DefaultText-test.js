/**
 * @format
 */

import 'react-native';
import React from 'react';
import render from 'react-test-renderer';
import DefaultText from '_atoms/DefaultText';

// Testing to see whether DefaultText component renders as it should be
test('DefaultText component snapshot', () => {
  const defaultTextSnap = render.create(<DefaultText />).toJSON();
  expect(defaultTextSnap).toMatchSnapshot();
});

// import {render} from '@testing-library/react-native';

// test('DefaultText component snapshot', () => {
//   const defaultTextSnap = render(<DefaultText />).toJSON();
//   expect(defaultTextSnap).toMatchSnapshot();
// });

// UPDATE SnapShot ==> npm test -- -u
