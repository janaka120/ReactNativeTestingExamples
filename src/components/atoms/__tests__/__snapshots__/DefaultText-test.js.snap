// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultText component snapshot 1`] = `
<Text
  multiline={true}
  numberOfLines={5}
  style={
    Array [
      Object {
        "color": "#594F4F",
        "fontSize": 29,
        "lineHeight": 39.875,
      },
      undefined,
    ]
  }
/>
`;
